FWCore/Services/src/CondorStatusUpdater.cc

```diff
@@ -0,0 +1,272 @@
+
+#include "CondorStatusUpdater.h"
+
+#include "DataFormats/Provenance/interface/ModuleDescription.h"
+#include "FWCore/ServiceRegistry/interface/ActivityRegistry.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/Registry.h"
+
+#include <fcntl.h>
+#include <unistd.h>
+#include <sys/wait.h>
+#include <spawn.h>
+#include <iostream>
+#include <cmath>
+#include <chrono>
+#include <sstream>
+
+using namespace edm::service;
+
+CondorStatusService::CondorStatusService(ParameterSet const& pset, edm::ActivityRegistry& ar)
+  :
+    m_debug(false),
+    m_events(0),
+    m_lumis(0),
+    m_runs(0),
+    m_files(0),
+    m_lastUpdate(0),
+    m_psetId(nullptr)
+{
+    m_shouldUpdate.clear();
+    if (pset.exists("debug"))
+    {
+        m_debug = true;
+    }
+    if (!isChirpSupported()) {return;}
+
+    ar.watchPostCloseFile(this, &CondorStatusService::filePost);
+    ar.watchPostEvent(this, &CondorStatusService::eventPost);
+    ar.watchPostGlobalEndLumi(this, &CondorStatusService::lumiPost);
+    ar.watchPostGlobalEndRun(this, &CondorStatusService::runPost);
+    ar.watchPostBeginJob(this, &CondorStatusService::beginPost);
+    ar.watchPostEndJob(this, &CondorStatusService::endPost);
+
+    if (pset.exists("updateIntervalSeconds"))
+    {
+        setUpdateInterval(pset.getUntrackedParameter<unsigned int>("updateIntervalSeconds"));
+    }
+    if (pset.exists("EMAInterval"))
+    {
+        m_emaInterval = pset.getUntrackedParameter<double>("EMAInterval");
+    }
+}
+
+
+void
+CondorStatusService::eventPost(StreamContext const& iContext)
+{
+    m_events++;
+    update();
+}
+
+
+void
+CondorStatusService::lumiPost(GlobalContext const&)
+{
+    m_lumis++;
+    update();
+}
+
+
+void
+CondorStatusService::runPost(GlobalContext const&)
+{
+    m_runs++;
+    update();
+}
+
+
+void
+CondorStatusService::filePost(std::string const & /*lfn*/, bool /*usedFallback*/)
+{
+    m_files++;
+    update();
+}
+
+
+void
+CondorStatusService::beginPost()
+{
+    ParameterSet const& processParameterSet = edm::getProcessParameterSet();
```

Comment:
I had several attempts to avoid using this function, but none were successful.

I would be quite happy if you had some guidance on how I could otherwise get access to the main source's PSet.